File: CatalogService/src/catalog/interfaces/icatalog.service.ts
import { CatalogInfo } from '../schemas/catalog-info.schema';
import { ItemInfo } from '../schemas/item-info.schema';
import { CatalogType } from '../enums/catalog-type.enum';

export interface ICatalogService {
  requestCatalog(catalogType: CatalogType): Promise<CatalogInfo[]>;
  requestSelection(itemId: number): Promise<ItemInfo>;
}

File: Broker/BrokerNCoordinator/src/main/java/com/example/BrokerNCoordinator/Service/Interface/ICatalogService.java
package com.example.BrokerNCoordinator.Service.Interface;

public interface ICatalogService {
//    requestCatalog (in catalogType, out catalogInfo)
//    requestSelection (in itemId, out itemInfo)
    public Object requestCatalog(Object catalogType);
    public Object requestSelection(Object itemId);
}
